src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,108 @@
+                if (words[0].equals("D")) {
+                    Deadline previousDeadline = new Deadline(words[2], words[3]);
+                    if (words[1].equals("1")) {
+                        previousDeadline.setIsDone();
+                    }
+                    tasks.add(previousDeadline);
+                } else if (words[0].equals("T")) {
+                    ToDo previousToDo = new ToDo(words[2]);
+                    if (words[1].equals("1")) {
+                        previousToDo.setIsDone();
+                    }
+                    tasks.add(previousToDo);
+                } else if (words[0].equals("E")) {
+                    Event previousEvent = new Event(words[2], words[3]);
+                    if (words[1].equals("1")) {
+                        previousEvent.setIsDone();
+                    }
+                    tasks.add(previousEvent);
+                }
+            }
+        } catch (FileNotFoundException e) {
+            e.printStackTrace();
+        }
+        return tasks;
+    }
+
+    /**
+     * Creates a duke.txt under data directory if the file does not exist at the first place.
+     * If the file exist, start loading. (Adding previous stored task in duke.txt into an ArrayList<tasks>)
+     *
+     * @throws IOException exception occurs when the file is not found
+     */
+    public void createFie() throws IOException {
+        File file = new File(this.pathName);
+        File dukeSave = new File(this.pathName + this.fileName);
+        if (!file.exists()) {
+            file.mkdirs();
+            dukeSave.createNewFile();
+        } else {
+            if (dukeSave.exists()) {
+                this.load();
+            } else {
+                dukeSave.createNewFile();
+            }
+        }
+
+    }
+
+    /**
+     * Stores newly added tasks and previous tasks from the TaskList in the duke.txt file
+     *
+     * @param tasks TaskList with newly added tasks and previous tasks
+     * @throws IOException exception occurs when the file is not found
+     */
+    public void storeFile(TaskList tasks) throws IOException {
+        FileWriter fw = new FileWriter("data/duke.txt");
+        for (int k = 0; k < tasks.size(); k++) {
+            Task currentTask = tasks.get(k);
+            String isDone = currentTask.isDone ? " 1 @ " : " 0 @ ";
+            if (currentTask instanceof ToDo) {
+                fw.write("T @" + isDone + currentTask.description + System.lineSeparator());
+            } else if (currentTask instanceof Deadline) {
+                String date = " @ " + ((Deadline) currentTask).by;
+                fw.write("D @" + isDone + currentTask.description + date + System.lineSeparator());
+            } else if (currentTask instanceof Event) {
+                String date = " @ " + ((Event) currentTask).at;
+                fw.write("D @" + isDone + currentTask.description + date + System.lineSeparator());
+            }
```

Comment:
Perhaps could handle the task message in each of the Task subclass?
For example, declaring a writeToFile method that returns a task message to be written to the file in the Task class. 
Then, overriding the method in Todo, Deadline and Event. 
Therefore, calling currentTask.writeToFile() will return the respective task message.

Reply:
Ah i see. Thank you